pkg/registry/apps/shorturl/register.go

```diff
@@ -0,0 +1,86 @@
+package shorturl
+
+import (
+	"fmt"
+	"github.com/grafana/grafana-app-sdk/app"
+	"github.com/grafana/grafana-app-sdk/simple"
+	"github.com/grafana/grafana/apps/shorturl/pkg/apis"
+	shorturlv0alpha1 "github.com/grafana/grafana/apps/shorturl/pkg/apis/shorturl/v0alpha1"
+	shorturlapp "github.com/grafana/grafana/apps/shorturl/pkg/app"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	grafanarest "github.com/grafana/grafana/pkg/apiserver/rest"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/apiserver/builder/runner"
+	"github.com/grafana/grafana/pkg/services/apiserver/endpoints/request"
+	"github.com/grafana/grafana/pkg/services/shorturls"
+	"github.com/grafana/grafana/pkg/setting"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type ShortURLAppProvider struct {
+	app.Provider
+	cfg     *setting.Cfg
+	service shorturls.Service
+	logger  log.Logger
+}
+
+func RegisterApp(
+	cfg *setting.Cfg,
+	service shorturls.Service,
+) *ShortURLAppProvider {
+	provider := &ShortURLAppProvider{
+		cfg:     cfg,
+		service: service,
+		logger:  log.New("shorturl::RawHandlers"),
+	}
+	appCfg := &runner.AppBuilderConfig{
+		OpenAPIDefGetter:    shorturlv0alpha1.GetOpenAPIDefinitions,
+		ManagedKinds:        shorturlapp.GetKinds(),
+		LegacyStorageGetter: provider.legacyStorageGetter,
+		CustomConfig: any(&shorturlapp.ShortURLConfig{
```

Comment:
This custom config was added only because I need the `cfg.AppURL` to calculate the returned `shortURL` in the mutation webhook